lib/puppet/node.rb

```diff
@@ -27,7 +27,10 @@ def initialize_from_hash(data)
     @name       = data['name']       || (raise ArgumentError, _("No name provided in serialized data"))
     @classes    = data['classes']    || []
     @parameters = data['parameters'] || {}
-    @environment_name = data['environment']
+    env_name = data['environment']
+    env_name = env_name.intern unless env_name.nil?
+    @environment_name = env_name
+    environment = env_name
```

Comment:
This will cause an environment lookup to occur and fail if the environment doesn't exist (due to https://github.com/thallgren/puppet/blob/6d42c0e6326b3293eb9513726edbfa32fa3a7892/lib/puppet/node.rb#L78). Is that a problem?